tests/unit/grains/test_disks.py

```diff
@@ -83,3 +83,43 @@ def test__windows_disks_retcode(self):
             result = disks._windows_disks()
             expected = {"SSDs": [], "disks": []}
             self.assertDictEqual(result, expected)
+
+    def test__linux_disks(self):
+        """
+        Test grains._linux_disks, normal return
+        Should return a populated dictionary
+        """
+
+        files = [
+            "/sys/block/asm!.asm_ctl_vbg0",
+            "/sys/block/dm-0",
+            "/sys/block/loop0",
+            "/sys/block/ram0",
+            "/sys/block/sda",
+            "/sys/block/sdb",
+            "/sys/block/vda",
+        ]
+        links = [
+            "../devices/virtual/block/asm!.asm_ctl_vbg0",
+            "../devices/virtual/block/dm-0",
+            "../devices/virtual/block/loop0",
+            "../devices/virtual/block/ram0",
+            "../devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0/block/sda",
+            "../devices/pci0000:35/0000:35:00.0/0000:36:00.0/host2/target2:1:0/2:1:0:0/block/sdb",
+            "../devices/pci0000L00:0000:00:05.0/virtio2/block/vda",
+        ]
+        contents = [
+            "1",
+            "1",
+            "1",
+            "0",
+            "1",
+            "1",
+            "1",
+        ]
+        with patch("glob.glob", MagicMock(return_value=files)), patch(
+            "salt.utils.path.readlink", MagicMock(side_effect=links)
+        ), patch("salt.utils.files.fopen", mock_open(read_data=contents)):
```

Comment:
```suggestion
       patch_glob = patch("glob.glob", autospec=True, return_value=files)
       patch_readlink = patch("salt.utils.path.readlink", autospec=True, side_effect=links)
       patch_fopen = patch("salt.utils.files.fopen", mock_open(read_data=contents), autospec=True)
       with patch_glob, patch_readlink, patch_fopen:
```

I'm pretty sure that's the right syntax. Anyway, the most important part of the change is `autospec=True` - basically what that does is ensure that the patched thing looks/behaves like the thing it's patching - so if the underlying API changes then this test will also (correctly) break. IIRC, patch will produce a MagicMock by default - you can just give it a return_value or side_effect.